contrib/src/json.rs

```diff
@@ -97,7 +97,13 @@ impl<T: DeserializeOwned> FromData for Json<T> {
         }
 
         let size_limit = request.limits().get("json").unwrap_or(LIMIT);
-        serde_json::from_reader(data.open().take(size_limit))
+        let mut buf = String::new();
+        data.open()
+            .take(size_limit)
+            .read_to_string(&mut buf)
+            .map_err(|e| { error_!("IO Error: {:?}", e); e })
+            .unwrap();
```

Comment:
I am not sure how to deal with this `unwrap()`.

Reply:
Unfortunately it doesn't look like Serde's error type can be created from an `io:Error`, so we'll have to change the error type here so that it returns an `io::Error` instead. We can then use [`impl From<serde::Error> for io::Error`](https://docs.serde.rs/serde_json/error/struct.Error.html#impl-From%3CError%3E). Alternatively, @dtolnay: would it be possible to add a `From<io::Error> for serde_json::error::Error` implementation?

Reply:
I would prefer not to have `From<io::Error> for serde_json::Error`. JSON errors are for Serde to create.

Reply:
Switched to `io::Error`